From b068d5a0a228029bb4aac50dc3f343971634f25e Mon Sep 17 00:00:00 2001
From: Sebastian Pipping <sebastian@pipping.org>
Date: Thu, 8 Oct 2026 13:03:52 +0200
Subject: [PATCH] lib: Add missing call to `poolDiscard` to function `doProlog`

.. after the call to `callStoreEntityValue`.
Follow-up to commit 30c2fc179ce5d2b1b1bae30bbe0dfddeac894e13.
---
 lib/xmlparse.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lib/xmlparse.c b/lib/xmlparse.c
index b3c65582..ea8fdc9f 100644
--- a/lib/xmlparse.c
+++ b/lib/xmlparse.c
@@ -5941,6 +5941,7 @@ doProlog(XML_Parser parser, const ENCODING *enc, const char *s, const char *end,
         if (parser->m_declEntity) {
           /* Detect and prevent signed integer overflow */
           if ((size_t)poolLength(&dtd->entityValuePool) > (size_t)INT_MAX) {
+            poolDiscard(&dtd->entityValuePool);
             return XML_ERROR_NO_MEMORY;
           }
           parser->m_declEntity->textPtr = poolStart(&dtd->entityValuePool);
